WIP: Added arm64 jobs in Travis-ci

odidev opened the pull request:
Updated travis.yml file to add ARM64 jobs in Travis-CI. Added support for python version 3.6, 3,7 and 3.8 for ARM64 jobs.
Resolves #1401.

minrk commented:
This is great! I think we can do fewer arm versions, just ZMQ= and ZMQ=bundled for python 3.7, 3.8 (4 total) and skip the tornado matrix items, which have no reason to behave differently on arm.

Since this is new, I think we can limit oldest ARM builds to 3.7, unless there's a really strong case that old 3.6 is very widely used on ARM.

For the wheel builds, I think we want to port over the logic from tools/tasks.py, which builds libzmq+libsodium from source, and we want to make sure the linux wheels built on CI do the same. Also for the wheel builds, we probably want to run the tests after installing from the wheel instead of running tests from the source directory prior to building an untested wheel.

odidev commented:
> I think we can do fewer arm versions, just ZMQ= and ZMQ=bundled for python 3.7, 3.8 (4 total), and skip the tornado matrix items, which have no reason to behave differently on arm.

@minrk As per your suggestion I have updated .travis.yml to run for python 3.7, 3.8 (4 total), and removed all the other jobs for arm64.

> I think we want to port over the logic from tools/tasks.py, which builds libzmq+libsodium from source, and we want to make sure the Linux wheels built on CI do the same.

Also, I have updated tools/tasks.py to create wheel for aarch64, now wheels are getting created successfully.

Please have a look and share your opinion.
